src/components/editor/MonacoEditor.tsx

```diff
@@ -22,35 +29,128 @@ function MonacoEditor({ disabled, value, path, height, onChange }: Props) {
+
+                if (!isEqual(currentEditorValue, serverUpdate)) {
+                    setHasServerChanges(serverUpdate);
+                } else {
+                    setHasServerChanges(null);
+                }
+            }
+        }
+
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [serverUpdate]);
+
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    const bar = useCallback(
+        debounce((val: string | undefined) => {
+            console.log('debounce', {
+                val,
+                curr: editorRef.current?.getValue(),
+            });
+            if (val && onChange) {
+                setIsChanging(true);
+                onChange(JSON.parse(val), path)
+                    .then(() => {
+                        setIsChanging(false);
+                    })
+                    .catch(() => {
+                        setIsChanging(false);
+                    });
             }
-        }, 1000),
+        }, 750),
+        []
+    );
+
+    const handlers = {
+        change: (val: string | undefined) => {
+            setIsChanging(true);
+            bar(val);
+        },
         mount: (editor: monacoEditor.editor.IStandaloneCodeEditor) => {
             editorRef.current = editor;
         },
+        merge: () => {
+            setShowServerDiff(!showServerDiff);
+        },
     };
 
     if (value) {
         return (
-            <Editor
-                height={`${heightVal}px`}
-                defaultLanguage="json"
-                theme={theme.palette.mode === 'light' ? 'vs' : 'vs-dark'}
-                defaultValue={intl.formatMessage({ id: 'common.loading' })}
-                value={JSON.stringify(value, null, 2)}
-                path={path}
-                options={{
-                    readOnly: disabled ? disabled : false,
-                }}
-                onMount={handlers.mount}
-                onChange={
-                    typeof onChange === 'function' ? handlers.change : undefined
-                }
-            />
+            <Paper sx={{ width: '100%' }} variant="outlined">
+                <Box
+                    sx={{
+                        minHeight: 20,
+                    }}
+                >
+                    <Stack
+                        direction="row"
+                        sx={{
+                            justifyContent: 'end',
+                            alignItems: 'center',
+                        }}
+                    >
+                        {hasServerChanges ? (
+                            <ServerDiff
+                                iconSize={ICON_SIZE}
+                                onMerge={handlers.merge}
+                            />
+                        ) : isChanging === null ? null : isChanging ? (
+                            <Saving iconSize={ICON_SIZE} />
+                        ) : (
+                            <Saved iconSize={ICON_SIZE} />
+                        )}
```

Comment:
If there are changes show the warning
If they haven't changed anything show nothing
If we are saving show the saving icon
If we saved show the saved icon

Still need to add an "Error Updating"